src/dag/get.js

```diff
@@ -0,0 +1,61 @@
+'use strict'
+
+const dagPB = require('ipld-dag-pb')
+const dagCBOR = require('ipld-dag-cbor')
+const promisify = require('promisify-es6')
+const CID = require('cids')
+const waterfall = require('async/waterfall')
+const block = require('../block')
+
+module.exports = (send) => {
+  return promisify((cid, path, options, callback) => {
+    if (typeof path === 'function') {
+      callback = path
+      path = undefined
+    }
+
+    if (typeof options === 'function') {
+      callback = options
+      options = {}
+    }
+
+    options = options || {}
+
+    if (CID.isCID(cid)) {
+      cid = cid.toBaseEncodedString()
+    }
+
+    if (typeof cid === 'string') {
+      const split = cid.split('/')
+      cid = split[0]
+      split.shift()
+
+      if (split.length > 0) {
+        path = split.join('/')
+      } else {
+        path = '/'
+      }
+    }
+
+    waterfall([
+      cb => {
+        send({
+          path: 'dag/resolve',
+          args: cid + '/' + path,
+          qs: options
+        }, cb)
+      },
+      (resolved, cb) => {
+        block(send).get(new CID(resolved['Cid']['/']), (err, blk) => cb(err, blk, resolved['RemPath']))
+      },
+      (blk, path, cb) => {
+        if (blk.cid.codec === 'dag-cbor') {
+          dagCBOR.resolver.resolve(blk, path, cb)
+        }
+        if (blk.cid.codec === 'dag-pb') {
+          dagPB.resolver.resolve(blk, path, cb)
+        }
+      }
+    ], callback)
```

Comment:
> Passing json directly instead of serializing it here would allow to support every encoding go side supports without additional work here.

Using json for all the serialization has a big drawback described here: https://github.com/ipfs/interface-ipfs-core/issues/81#issuecomment-277271941

tl;dr; JSON doesn't support Buffers, which makes it really hard to distinguish a Buffer from a String, you can see that we do a cast to Buffer every single time in `object get` just because we know that Data in the protobuf is always binary. This is the tl;dr; but please do read the whole thread :)

> The same is the case for get with the difference that using go-ipfs dag get doesn't return remainderPath(I don't think there is any case where it wouldn't be empty)

`remainderPath` should return any path that the node wasn't able to resolve either because it couldn't find more nodes or because it didn't have the right parser for it.